resources/aws/policy.go

```diff
@@ -54,6 +54,52 @@ const (
 				"Effect": "Allow",
 				"Action": "s3:GetObject",
 				"Resource": "arn:aws:s3:::%s/*"
+			}
+		]
+	}`
+	MasterPolicyType          = "master"
+	WorkerPolicyDocumentTempl = `{
+		"Version": "2012-10-17",
+		"Statement": [
+			{
+				"Effect": "Allow",
+				"Action": "ec2:Describe*",
+				"Resource": "*"
+			},
+			{
+				"Effect": "Allow",
+				"Action": "ec2:AttachVolume",
+				"Resource": "*"
+			},
+			{
+				"Effect": "Allow",
+				"Action": "ec2:DetachVolume",
+				"Resource": "*"
+			},
+			{
+				"Effect": "Allow",
+				"Action": "kms:Decrypt",
+				"Resource": %q
+			},
+			{
+				"Effect": "Allow",
+				"Action": [
+					"s3:GetBucketLocation",
+					"s3:ListAllMyBuckets"
+				],
+				"Resource": "*"
+			},
+			{
+				"Effect": "Allow",
+				"Action": [
+					"s3:ListBucket"
+				],
+				"Resource": "arn:aws:s3:::%s"
```

Comment:
Yes I think we should move to golang templates. The templates will need more logic when we make the ECR permissions configurable. We may also need to add templates for Cloud Formation.